SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+        the edit project dialog will send the information again to this endpoint to fix the project information.
+        """
+        input_data = self.get_json_body()
+        print(f"EditProjectHandler = {input_data}")
+
+        corrupted = False
+        if "corrupted" in input_data.keys():
+            corrupted = input_data["corrupted"]
+
+        old_name = ""
+        if "old_name" in input_data.keys():
+            old_name = input_data["old_name"]
+
+        old_stack = ""
+        if "old_stack" in input_data.keys():
+            old_stack = input_data["old_stack"]
+
+        old_platform = ""
+        if "old_platform" in input_data.keys():
+            old_platform = input_data["old_platform"]
+
+        old_release = ""
+        if "old_release" in input_data.keys():
+            old_release = input_data["old_release"]
+
+        old_userscript = ""
+        if "old_userscript" in input_data.keys():
+            old_userscript = input_data["old_userscript"]
+
+        name = input_data["name"]
+        stack = input_data["stack"]
+        platform = input_data["platform"]
+        release = input_data["release"]
+        user_script = input_data["user_script"]
+
+        project_dir = os.environ["HOME"] + "/SWAN_projects/" + name
+        if old_name != name:
+            try:
+                old_project_dir = os.environ["HOME"] + \
+                    "/SWAN_projects/" + old_name
+                os.rename(old_project_dir, project_dir)
+            except Exception as msg:
+                data = {"status": False, "project_dir": f"SWAN_projects/{old_name}",
+                        "msg": f"Error editing project folder {old_name},  traceback: {msg}"}
+                # this will stop the execution here, it's the same for the next exceptions.
+                self.finish(json.dumps(data))
+                return
+
+        if old_userscript != user_script:
+            userscript_file = project_dir + os.path.sep + '.userscript'
+            try:
+                with open(userscript_file, 'w') as f:
+                    f.write(user_script)
+                    f.close()
+            except Exception as msg:
+                data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                        "msg": f"Error editing .userscript for project {name},  traceback: {msg}"}
+                self.finish(json.dumps(data))
+                return
+
+        if stack != old_stack or platform != old_platform or release != old_release or corrupted:
+            swan_project_file = project_dir + os.path.sep + '.swanproject'
+            swan_project_content = {'stack': stack, 'release': release,
+                                    'platform': platform}
+            kernel_dir = project_dir + "/.local/share/jupyter/kernels"
+
+            # removing old native kernels for python only(this is generated by us)
+            if os.path.exists(kernel_dir + "/python2"):
```

Comment:
Where are these native kernels regenerated for the new stack?

Reply:
Why are we deleting them??? Can't we just hide them? I'm not in favor of deleting users' stuff, especially when they can be using it from somewhere else!

Reply:
1) The native kernels are generated inside the project folder
2) When the user edits the project for a new stack, those kernels that we generate are eliminated and replaced by the new ones, if the user wants the previous kernels, it is only to select the previous stack.